predicators/ground_truth_models/kitchen/nsrts.py

```diff
@@ -0,0 +1,117 @@
+"""Ground-truth NSRTs for the Kitchen environment."""
+
+from typing import Dict, Sequence, Set
+
+import numpy as np
+
+from predicators.ground_truth_models import GroundTruthNSRTFactory
+from predicators.structs import NSRT, Array, GroundAtom, LiftedAtom, Object, \
+    ParameterizedOption, Predicate, State, Type, Variable
+
+
+class KitchenGroundTruthNSRTFactory(GroundTruthNSRTFactory):
+    """Ground-truth NSRTs for the Kitchen environment."""
+
+    @classmethod
+    def get_env_names(cls) -> Set[str]:
+        return {"kitchen"}
+
+    @staticmethod
+    def get_nsrts(env_name: str, types: Dict[str, Type],
+                  predicates: Dict[str, Predicate],
+                  options: Dict[str, ParameterizedOption]) -> Set[NSRT]:
+        # Types
+        gripper_type = types["gripper"]
+        object_type = types["obj"]
+
+        # Objects
+        gripper = Variable("?gripper", gripper_type)
+        obj = Variable("?obj", object_type)
+        obj2 = Variable("?obj2", object_type)
+
+        # Predicates
+        At = predicates["At"]
+        On = predicates["On"]
+        OnTop = predicates["OnTop"]
+
+        # Options
+        MoveTo = options["Move_delta_ee_pose"]
+
+        nsrts = set()
+
+        # Samplers
+        def moveto_sampler(state: State, goal: Set[GroundAtom],
+                           _rng: np.random.Generator,
+                           objs: Sequence[Object]) -> Array:
+            del goal, _rng
+            _, obj = objs
+            ox = state.get(obj, "x")
+            oy = state.get(obj, "y")
+            oz = state.get(obj, "z")
+            if obj.name == 'knob2':
+                return np.array([ox - 0.05, oy + 0.05, oz - 0.05, 0.0],
+                                dtype=np.float32)
+            if obj.name == 'knob3':
+                return np.array([ox - 0.1, oy + 0.02, oz - 0.1, 0.0],
+                                dtype=np.float32)
+            if obj.name == 'kettle':
+                return np.array([ox + 0.15, oy - 0.4, oz - 0.15, 0.0],
+                                dtype=np.float32)
+            return np.array([ox, oy, oz, 0.0], dtype=np.float32)
+
+        def push_sampler(state: State, goal: Set[GroundAtom],
+                         _rng: np.random.Generator,
+                         objs: Sequence[Object]) -> Array:
+            del state, goal, _rng
+            if len(objs) == 2:
+                _, obj = objs
+            else:
+                assert len(objs) == 3
+                _, obj, _ = objs
+            if obj.name == 'knob2':
```

Comment:
if we want to learn these samplers, will we need to add some features to distinguish between these different objects? or are their positions sufficient?